### PR TITLE
reopen hub on crash

### PR DESCRIPTION
**Pull request by amirbilu**

### CODE REVIEW

File: src/binary/requests/requests.ts [hunk 1]
```diff
@@ -166,3 +167,9 @@ export async function saveSnippet(
 ): Promise<SaveSnippetResponse | null | undefined> {
   return tabNineProcess.request({ SaveSnippet: args });
 }
+
+export function onTabnineProcessRestart(
```
Comment (dimacodota) [inline src/binary/requests/requests.ts:171]: not related here, move it

**Review by dimacodota — APPROVED** [review]
see my comment, lgtm

File: src/binary/Binary.ts [hunk 2]
```diff
@@ -1,14 +1,18 @@
 import * as child_process from "child_process";
+import { EventEmitter } from "stream";
```
Comment (assapir) [inline src/binary/Binary.ts:2]: Isn't that imported from events?

**Review by assapir — APPROVED** [review]
👏🏻

**Pull request merged**